### ISSUE TITLE
Plotting labels to *.jpg

### ISSUE THREAD

**Issue by yyq19990828** [ultralytics/ultralytics#21174]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

When merging multiple datasets in dataset.yaml, the terminal statistics and plot image only show the first dataset, which inevitably makes me confused whether the dataset is fully loaded into the dataloader

### Environment

```
Ultralytics 8.3.158 🚀 Python-3.8.10 torch-1.14.0a0+410ce96 CUDA:0 (NVIDIA GeForce RTX 3090, 24252MiB)
Setup complete ✅ (112 CPUs, 251.5 GB RAM, 2813.3/3666.5 GB disk)

OS                  Linux-5.15.0-139-generic-x86_64-with-glibc2.29
Environment         Docker
Python              3.8.10
Install             git
Path                /workspace/ultralytics_ori/ultralytics
RAM                 251.51 GB
Disk                2813.3/3666.5 GB
CPU                 Intel Xeon Gold 6238R 2.20GHz
CPU count           112
GPU                 NVIDIA GeForce RTX 3090, 24252MiB
GPU count           2
CUDA                11.8
```

### Minimal Reproducible Example

![Image](https://github.com/user-attachments/assets/142a2339-605e-4259-bd45-757d07b6f3af)

### Additional

By the way, the generated cache file will also be stored in the first dataset. When my other datasets change, the code will still read this cache and will not update the cache file.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yyq19990828, thank you for bringing this up and for your detailed report! 🚀 This is an automated response to help get you the information you need quickly—an Ultralytics engineer will also assist you soon.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for guidance, especially if you’re merging datasets or working with custom data. For more context and examples, see our [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage guides.

Since you are reporting a 🐛 Bug, please ensure your minimal reproducible example (MRE) contains all necessary details, and see our [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) guide for reference. This will help us debug the issue efficiently.

If you’re working with custom training or dataset merging, please provide as much information as possible, including dataset YAMLs, logs, and sample images, and review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community for more support:
- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions on [Discourse](https://community.ultralytics.com/)
- Share and learn on our [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

Please upgrade to the latest `ultralytics` package (and dependencies in [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to make sure your issue is not already resolved:

```bash
pip install -U ultralytics
```

## Environments

YOLO runs in these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for your report! An Ultralytics engineer will review your issue soon.

**Comment by Y-T-G** [comment]
Which terminal statistics are you referring to?

Also by plot are you referring to the batch plots like `train_batchX.jpg`?

**Comment by yyq19990828** [comment]
I mean the labels.jpg. Also I found that the potential cause of this error may be that the image folder of one of my datasets uses a soft link, pointing to ./image from somewhere else, which causes the code to search for labels in the original path of this soft link.

> Which terminal statistics are you referring to?
> 
> Also by plot are you referring to the batch plots like `train_batchX.jpg`?

**Comment by Y-T-G** [comment]
I can't reproduce the issue with `labels.jpg`. It seems to correctly show the instances from all the datasets. But I can reproduce the issue with `cache` file using the first dataset folder.

**Comment by Y-T-G** [comment]
But if you add more images, the cache is regenerated because the hash values wouldn't match with older cache. So it's not an issue.
